preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayers.groovy

```diff
@@ -74,21 +77,24 @@ IProcess extractAndCreateGISLayers(){
                 /**
                  * Extract the OSM file from the envelope of the geometry
                  */
-                def geomAndEnv = OSMTools.Utilities.buildGeometryAndZone(geom, epsg, distance, datasource)
-                epsg = geomAndEnv.geom.getSRID()
+                def envelope = GeographyUtils.expandEnvelopeByMeters(geom.getEnvelopeInternal(), distance)
+
+                //Find the best utm zone
+                //Reproject the geometry and its envelope to the UTM zone
+                def con = datasource.getConnection();
+                def interiorPoint = envelope.centre()
+                def epsg = SFSUtilities.getSRID(con, interiorPoint.y as float, interiorPoint.x as float)
+                Geometry geomUTM = ST_Transform.ST_Transform(con, geom, epsg)
+                Geometry tmpGeomEnv = geom.getFactory().toGeometry(envelope)
```

Comment:
If you can, it's better to use `def` instead of explicit typing.

Reply:
done